gateway/src/main/java/io/camunda/zeebe/gateway/EndpointManager.java

```diff
@@ -276,35 +274,32 @@ public void topology(final ServerStreamObserver<TopologyResponse> responseObserv
     final TopologyResponse.Builder topologyResponseBuilder = TopologyResponse.newBuilder();
     final BrokerClusterState topology = topologyManager.getTopology();
 
-    if (topology == null) {
-      final StatusRuntimeException error =
-          Status.UNAVAILABLE.augmentDescription("No brokers available").asRuntimeException();
-      responseObserver.onError(error);
-      return;
-    }
-
-    topologyResponseBuilder
-        .setClusterSize(topology.getClusterSize())
-        .setPartitionsCount(topology.getPartitionsCount())
-        .setReplicationFactor(topology.getReplicationFactor());
-
     final String gatewayVersion = VersionUtil.getVersion();
     if (gatewayVersion != null && !gatewayVersion.isBlank()) {
       topologyResponseBuilder.setGatewayVersion(gatewayVersion);
     }
 
     final ArrayList<BrokerInfo> brokers = new ArrayList<>();
 
-    topology
-        .getBrokers()
-        .forEach(
-            brokerId -> {
-              final Builder brokerInfo = BrokerInfo.newBuilder();
-              addBrokerInfo(brokerInfo, brokerId, topology);
-              addPartitionInfoToBrokerInfo(brokerInfo, brokerId, topology);
-
-              brokers.add(brokerInfo.build());
-            });
+    if (topology == null) {
+      topologyResponseBuilder.setClusterSize(-1).setPartitionsCount(-1).setReplicationFactor(-1);
+    } else {
```

Comment:
:wrench: I would collapse this as `if (topology != null)` and have no else. By default the builder has 0 for size, count, and factor. Is it important for us to have -1? :thinking:

Reply:
I was having trouble deciding between 0 and -1. In the end I figured -1 would be a better indication that the real amounts are unknown. If you disagree I'm happy to change it.

Reply:
[0 is the default value for protobuf](https://developers.google.com/protocol-buffers/docs/proto3#default).

So if we have -1, then we still have two different cases:

1. When no topology is known, we return -1 for size, count, and factor.
2. When there is one but we haven't managed to connect to anyone, then we will have 0.

Did I missing anything? Do we get anything from distinguishing the cases? I don't think so, but like I said, maybe I'm forgetting something. If not, then I would align the values to the defaults.